app/controllers/admin/contacts_controller.rb

```diff
@@ -47,12 +48,14 @@ def create
     end
 
     def destroy
```

Comment:
If could make this is a bit more generic around authorization…

```suggestion
    def destroy
      association = ResourceContact.find_by(contact_id: params[:id], resource_id: params[:location_id], resource_type: Location.to_s).includes(:contact)
      authorize(association)
      assocation.destroy
      # contact may have to be pulled off the association record first here. can't remember if it is available post-destroy, though I think it would be.
      redirect_to(admin_location_url(association.resource_id), notice: "Contact '#{association.contact.name}' was successfully deleted.")
    end
```
Thinking aloud some more. This might be more trouble than worth.